src/vm_cpu.rs

```diff
@@ -869,6 +869,10 @@ impl<'vcpu, 'pages, 'host, T: GuestStagePagingMode> ActiveVmCpu<'vcpu, 'pages, '
 
     /// Sets one of the vCPU's general-purpose registers.
     pub fn set_gpr(&mut self, gpr: GprIndex, value: u64) {
+        if gpr == GprIndex::Zero {
```

Comment:
Personally, I would put this on the `GeneralPurposeRegisters` methods (`set_reg`, or even returning zero on `reg` for zero) itself. Afterall AFAIU the zero register should be zero always. So preventing it to be set at the type structure would just avoid looking like a corner special case.

Reply:
Makes sense. :+1: Thanks. Fixed.